### ISSUE TITLE
Total Price with TAX issue

### ISSUE THREAD

**Issue by mattusik** [Crinsane/LaravelShoppingcart#164]
Hi, I run into issue with prices and Tax. 

Correct me If I'm wrong, I'm adding product into Cart with price without Tax, like this:
`Cart::add($product->id, $product->name, 1, $product->price, $options);`

All of vars `CartItem::subtotal` total price without Tax, `CartItem::price` price without Tax for 1 product and `CartItem::tax` total Tax value  (I'm just missing `CartItem::total` for total price with Tax for product)

so, everything is ok for now, but when I try to get totals from whole Cart 
`Cart::total()` gives me total price for whole Cart without Tax and `Cart::subtotal()` is trying to recalculate price without Tax from value which is without tax right now, which is wrong.

I've updated your total method in Cart class, by adding tax to price:
```
public function total($decimals = 2, $decimalPoint = '.', $thousandSeperator = ',')
{
    $content = $this->getContent();

    $total = $content->reduce(function ($subTotal, CartItem $cartItem) {
        return $subTotal + ($cartItem->qty * ($cartItem->price + $cartItem->tax));
    }, 0);

    return number_format($total, $decimals, $decimalPoint, $thousandSeperator);
}
```
and everything started working

**Comment by Crinsane** [comment]
What I think the problem is, is this. You're expecting the `price` attribute of a CartItem to be the price without tax, where the packages uses the `price` as the price of an CartItem with the tax.

So when you do `Cart::add('123abc', 'My Product', 1, 10.00);` You're adding an item to the cart that has a price of 10.00 for the customer. Say the taxrate is 21%, this means that 2.10 of this price is tax. So the price without tax for the product would be 7.90.

So this logic would give you:

```php
// Lets work with 19% tax
Cart::add('123abc', 'My Product', 2, 10.00); // Add two products each 10 dollar *with tax*

$cartItem->price == 10.00 // Price of the product *with tax*
$cartItem->subTotal == 20.00 // 2 * 10.00, total price of given quantity of the product *with tax*
$cartItem->tax == 1.90 // Tax for 1 product for the given tax rate

Cart::subtotal() == 16.20 // Total of the cart, minus the tax (2 * 10.00 - 2 * 1.90)
Cart::tax() == 3.80 // Total tax of the cart (2 * 1.90)
Cart::total() == 20.00 // Total of the cart including tax
```

**Comment by mattusik** [comment]
Ok, then. So, your calculation of tax for CartItem is wrong, you can't calculate tax from **price with tax** in way like you're doing it.

Let's say that price with tax is 10, like you said before. 

Then your tax calculation `$this->price * ($this->taxRate / 100);` returned tax = 1.9, which is completely wrong because price without tax by you is now `10 - 1.9 = 8.1`

I updated tax calculation to this `$this->price / ( 1 + ($this->taxRate / 100));` and tax is now 1.5966..., when I used new tax value for calculation price without tax I've got `10 - 1.5966 = 8.4034`

Let's prove it by calculating price with tax from price without tax. I assume that we can do it in this way `priceWithoutTax * (1 + (tax/100))`.

`8.1 * (1 + (19 / 100)) = 9.639` <-- your tax
`8.4034 * (1 + (19 / 100)) = 10` <-- mine tax

From what I know, all over the world tax is calculated from price without tax, not vice versa.

**Comment by Crinsane** [comment]
Hmm... I think you're correct indeed, there is something wrong. I'll reopen this and lets see how we can fix this :)

**Comment by Crinsane** [comment]
Could you send a pull request with the update? I indeed made a mistake in my calculations... stupid...

**Comment by Crinsane** [comment]
Fixed by #165
